lib/Dialect/Verif/Transforms/StripContracts.cpp

```diff
@@ -0,0 +1,35 @@
+//===- StripContracts.cpp -------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/Verif/VerifOps.h"
+#include "circt/Dialect/Verif/VerifPasses.h"
+#include "mlir/Pass/Pass.h"
+#include "llvm/Support/Debug.h"
+
+namespace circt {
+namespace verif {
+#define GEN_PASS_DEF_STRIPCONTRACTSPASS
+#include "circt/Dialect/Verif/Passes.h.inc"
+} // namespace verif
+} // namespace circt
+
+using namespace mlir;
+using namespace circt;
+using namespace verif;
+
+namespace {
+struct StripContractsPass
+    : public verif::impl::StripContractsPassBase<StripContractsPass> {
+  void runOnOperation() override {
+    getOperation()->walk([](ContractOp op) {
+      op.replaceAllUsesWith(op.getInputs());
+      op.erase();
+    });
```

Comment:
Pretty degenerate case but does this work in the following case?
```
%0 = verif.contract %1 : i32 {}
%1 = verif.contract %0 : i32 {}
```

Reply:
Good thinking. Should work. Let me add this to the sets.

Reply:
Actually needed a tweak to avoid an assert in `replaceAllUsesWith` and to allow `op.erase` to work if the operation is cycle.